### PR TITLE
don't show all the goals on mobile

### PR DESCRIPTION
**Pull request by xdamman**
Before:

After:

### CODE REVIEW

File: src/components/GoalsCover.js [hunk 1]
```diff
@@ -61,6 +61,11 @@ class GoalsCover extends React.Component {
     });
   }
 
+  componentDidMount() {
+    const state = this.populateGoals();
```
Comment (znarf) [inline src/components/GoalsCover.js]: I don't like this pattern. Why not have `getGoals()` and juste query that in `render()`?

File: src/components/GoalsCover.js [hunk 2]
```diff
@@ -90,6 +95,9 @@ class GoalsCover extends React.Component {
       });
     }
 
+    const maxGoalsToShow =
+      window.screen.availWidth < 400 ? this.goals.length + 1 : 10;
```
Comment (znarf) [inline src/components/GoalsCover.js]: Could we use the window size? And recompute when the window size change?

File: src/components/GoalsCover.js (hunk 1 above)
Comment (xdamman) [reply src/components/GoalsCover.js]: Render gets called on the backend and we don’t have the window size.

I don’t like that pattern either but my goal here was not to refactor, but just to show less goals on mobile.

File: src/components/GoalsCover.js (hunk 2 above)
Comment (xdamman) [reply src/components/GoalsCover.js]: That’s the window size despite the name.

Recomputing when it changes would be nice. Let me look into this.

Comment (xdamman) [reply src/components/GoalsCover.js]: Actually you are right. It’s  the width of the screen, not the window. Fixing this.

**Pull request merged**